orchestrator/src/compression/squash.rs

```diff
@@ -0,0 +1,412 @@
+                if value != Felt::ZERO {
+                    storage_entries.push(StorageEntry { key, value });
+                }
+            }
+        }
+        Some(pre_range_block) => {
+            // Check if contract existed at pre-range block
+
+            let contract_existed = if contract_addr == Felt::ONE || contract_addr == Felt::TWO {
+                false
+            } else {
+                check_contract_existed_at_block(provider, contract_addr, pre_range_block).await?
+            };
+
+            if contract_existed {
+                // Process storage entries only for an existing contract
+                let results: Vec<_> = stream::iter(storage_map)
+                    .map(|(key, value)| {
+                        let provider = Arc::clone(provider);
+                        async move {
+                            let pre_range_value =
+                                check_pre_range_storage_value(&provider, contract_addr, key, pre_range_block).await?;
+
+                            Ok::<_, JobError>((key, value, pre_range_value))
+                        }
+                    })
+                    .buffer_unordered(MAX_CONCURRENT_GET_STORAGE_AT_CALLS)
+                    .collect()
+                    .await;
+
+                // Process results
+                for result in results {
+                    let (key, value, pre_range_value) = result?;
+                    if pre_range_value != value {
+                        storage_entries.push(StorageEntry { key, value });
+                    }
+                }
+            } else {
+                // Contract didn't exist, filter non-zero values
+                for (key, value) in storage_map {
+                    if value != Felt::ZERO {
+                        storage_entries.push(StorageEntry { key, value });
+                    }
+                }
+            }
+        }
+    }
+
+    if !storage_entries.is_empty() {
+        Ok(Some(ContractStorageDiffItem { address: contract_addr, storage_entries }))
+    } else {
+        Ok(None)
+    }
+}
+
+/// This function tells if the contract existed at the given block number
+pub async fn check_contract_existed_at_block(
+    provider: &Arc<JsonRpcClient<HttpTransport>>,
+    contract_address: Felt,
+    block_number: u64,
+) -> Result<bool, JobError> {
+    Ok(get_class_hash_at(provider, block_number, contract_address).await?.0)
+}
+
+/// This function returns the class hash of a contract at a given block number
+/// If it exists, it returns the class hash along with `exists=true`
+/// If it doesn't exist, it returns the zero-class hash along with `exists=false`
+/// It retries up to `MAX_GET_STORAGE_AT_CALL_RETRY` times
+/// This function does not return error when the contract does not exist
+pub async fn get_class_hash_at(
+    provider: &Arc<JsonRpcClient<HttpTransport>>,
+    block_number: u64,
+    contract_address: Felt,
+) -> Result<(bool, Felt), JobError> {
```

Comment:
```suggestion
) -> Result<Option<Felt>, JobError> {
```

TODO: check if this is possible

Reply:
Done